clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java

```diff
@@ -90,6 +91,7 @@ public void configure(Map<String, ?> configs, String contextName, Configuration
         this.minTimeBeforeRelogin = (Long) configs.get(SaslConfigs.SASL_KERBEROS_MIN_TIME_BEFORE_RELOGIN);
         this.kinitCmd = (String) configs.get(SaslConfigs.SASL_KERBEROS_KINIT_CMD);
         this.serviceName = getServiceName(configs, contextName, configuration);
+        this.callbackHandler = callbackHandler;
```

Comment:
We could add a protected method `loginCallbackHandler()` to AbstractLogin (similar to `contextName()`) instead of storing it here as well.

Reply:
We can extract `loginCallbackHandler()` into AbstractLogin, true. I was trying to limit the change to the Kerberos code because I don't have a way to verify the impact across all other login methods.

I will create JIRA, no problem. 

I'm not immediately sure how to create a test for it within the Kafka code base - I don't see the live Kerberos tests in the tests/ tree. Will look into that.

Reply:
I have moved the loginHandler to AbstractLogin and added a test.

I've sent an email to private@kafka.apache.org to create a JIRA account for me to open a security issue.